blockchain/My-Wallet-V3-iOS
# chore(Project): update project for Xcode 10 support

## Objective

Add support for Xcode 10.

## Description

There are some important changes required for the project to compile on Xcode 10.
Also note that this pull request is the foundation for adding CI/CD support to this project.

## How to Test

This pull request requires testing with both Xcode version 9 & 10. It is crucial that these changes are tested using Xcode 9 first, to ensure backwards compatibility. The following steps walk you through the upgrade process:

0. Close Xcode (duh 🤦‍♂️).
1. Upgrade Xcode to version 10 via the App Store (leave the project closed to avoid any side effects).
2. Once complete, run `pod cache clean --all` to remove all cached pods.
3. Run `pod install` to install the latest version of each pod.
4. In the latest version of the Onfido SDK, several dependencies have been deprecated to add support for Xcode 10. To compensate for these changes, you'll need to update all of your `Pods-Blockchain.* *.xcconfig` files, located in the `Pods` directory.

Flags to be updated: `OTHER_CFLAGS` and `OTHER_LDFLAGS`.
```
OTHER_CFLAGS = $(inherited) -iquote "${PODS_CONFIGURATION_BUILD_DIR}/Alamofire/Alamofire.framework/Headers" -iquote "${PODS_CONFIGURATION_BUILD_DIR}/Charts/Charts.framework/Headers" -iquote "${PODS_CONFIGURATION_BUILD_DIR}/PhoneNumberKit/PhoneNumberKit.framework/Headers" -iquote "${PODS_CONFIGURATION_BUILD_DIR}/RxCocoa/RxCocoa.framework/Headers" -iquote "${PODS_CONFIGURATION_BUILD_DIR}/RxSwift/RxSwift.framework/Headers" -iquote "${PODS_CONFIGURATION_BUILD_DIR}/Starscream/Starscream.framework/Headers"

OTHER_LDFLAGS = $(inherited) -framework "AVFoundation" -framework "Alamofire" -framework "Charts" -framework "Foundation" -framework "Onfido" -framework "PhoneNumberKit" -framework "RxCocoa" -framework "RxSwift" -framework "Starscream" -framework "UIKit"
```
5. Open `Blockchain.xcworkspace` and clean the project (`Product -> Clean Build Folder`).
7. Navigate to `Xcode -> Preferences... -> Locations` and click on the small arrow icon next the folder path for your *Derived Data* to reveal it in Finder. **DELETE ALL THE THINGS**! (Should you have any further problems related to caching, run the following set of commands):
```
killall Xcode
xcrun -k
xcodebuild -alltargets clean
rm -rf "$(getconf DARWIN_USER_CACHE_DIR)/org.llvm.clang/ModuleCache"
rm -rf "$(getconf DARWIN_USER_CACHE_DIR)/org.llvm.clang.$(whoami)/ModuleCache"
rm -rf ~/Library/Developer/Xcode/DerivedData/*
rm -rf ~/Library/Caches/com.apple.dt.Xcode/*
```
8. Build the project and cross your fingers 🤞
9. If the project still does not compile, you might need to update the pods master repository by running `pod install --repo-update`.

## Merge Checklist
